src/components/provider.go

```diff
@@ -0,0 +1,71 @@
+package components
+
+import (
+	"fmt"
+	"keboola-as-code/src/model"
+	"sort"
+	"sync"
+)
+
+type remoteProvider interface {
+	GetComponent(componentId string) (*model.Component, error)
+}
+
+type Provider struct {
```

Comment:
Namiesto `remote.ComponentsCache` je `components.Provider` ... tym ze som pouzil interface `remoteProvider`, tak uz tu nie je zavislost na `remote.StorageAPI`.

Problem bol, ze informacie o komponentach potrebujem aj v `local` package, ... a v `remote` zase potrebujem `local` na ulozenie noveho ID na disk. Takze vznikala zavislost `local` -> `remote` -> `local` ... a teraz to je `local` -> `components`, ... co je viac logicke a nerobi to problemy.